astrocloud release 1.4.0 (#602)
astronomer/docs: astro/cli-reference/astrocloud-deployment-variable-list.md

---
sidebar_label: "astrocloud deployment variable list"
title: "astrocloud deployment variable list"
id: astrocloud-deployment-variable-list
description: Reference documentation for astrocloud deployment variable list.
---

## Description

> List a Deployment's environment variables and save them to a local `.env` file. If an environment variable value is set as secret, the CLI will list only its key.

Reviewer: ```suggestion

For a given Deployment on Astro, list its running environment variables in your terminal. To test these environment variables in a local environment, you can also use this command to save them in a local `.env` file and avoid having to manually copy them.

If an environment variable value is set as secret, the CLI will list only its key.
```

## Usage

```sh
astrocloud deployment variable list
```

## Options

| Option                         | Description                                                                            | Possible Values                                                                |
| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
| `-d`,`--deployment-id`           |    The Deployment for which to list environment variables                                                | Any valid Deployment ID |

> | `-e`,`--env`                  | The directory where the `.env` file will be generated (default is your current directory)                                                                 | Any valid filepath       |

Reviewer: ```suggestion
| `-e`,`--env`                  | The directory where the `.env` file will be generated if it does not already exist. The default is your current directory                                                                 | Any valid filepath       |
```

> | `-k`,`--key`             | Only list the variable with the provided key                                                  | Any string |

Reviewer: ```suggestion
| `-k`,`--key`             | List only the environment variable associated with this key. If not specified, all environment variables are listed                                                  | Any string |
```

> | `-s`,`--save`    | Save environment variables to an environment file               |`` |

Reviewer: ```suggestion
| `-s`,`--save`    | Save environment variables to a local `.env` file               |`` |
```

| `-w`,`--workspace-id`          | List environment variables for a Deployment outside of your current Workspace           | Any valid Workspace ID                                                         |

## Examples

```sh

> # Save environment variables from a specific Directory as an `.env` file in your current directory
> $ astrocloud deployment variable list cl03oiq7d80402nwn7fsl3dmv --save

Reviewer: ```suggestion
# Save environment variables currently running on an Astro Deployment to a `.env` file in your Astro project
$ astrocloud deployment variable list cl03oiq7d80402nwn7fsl3dmv --save
```

> # Save a single variable to an `.env` file outside of your current directory

Reviewer: ```suggestion
# Save only a single environment variable from a Deployment on Astro to a `.env` file that is outside of your current directory
```

> $ astrocloud deployment variable list cl03oiq7d80402nwn7fsl3dmv --save --env /users/myaccount/documents/myproject --key AIRFLOW_VAR_MY_VAR

Reviewer: ```suggestion
$ astrocloud deployment variable list cl03oiq7d80402nwn7fsl3dmv --key AIRFLOW_VAR_MY_VAR --save --env /users/documents/my-astro-project/.env
```
